httpx/_dispatch/asgi.py

```diff
@@ -76,23 +95,27 @@ async def request(
         status_code = None
         response_headers = None
         body_parts = []
+        request_complete = False
         response_started = False
-        response_complete = False
+        response_complete = create_event()
 
         headers = [] if headers is None else headers
         stream = ByteStream(b"") if stream is None else stream
 
         request_body_chunks = stream.__aiter__()
 
         async def receive() -> dict:
-            nonlocal response_complete
+            nonlocal request_complete, response_complete
 
-            if response_complete:
+            if request_complete:
+                # Simulate blocking until the response is complete and then disconnect
```

Comment:
I'd tweak this comment. We're not "simulating" blocking until the response is complete, we *are* blocking until the response is complete. :)

Reply:
Took out "streaming" and then found the comment a bit redundant 🤔